### PR TITLE
Manually dispatch a change event after updating a field

### PR DESCRIPTION
**Pull request by hackel**
Fix for #638.

My editor stripped trailing spaces.  The only actual change here is the addition on line 172.

### CODE REVIEW

**Comment by bbbco** [comment]
FYI, I pulled your one line change in to my extension and tested it out. This seems to address the issues I have seen where the text fields don't trigger the onChange event in at least a couple of sites.

**Comment by bdurrer** [comment]
While the many changes to the formatting are nice, they make the PR hard to read.
The only _real_ change is on line 172.

Would love to see this PR in the official KeeFox

**Comment by bbbco** [comment]
@hackel I'd recommend fixing the formatting for this PR to get through

**Comment by hackel** [comment]
@bbbco Fix what formatting?  Surely you don't want me to put the extra trailing whitespace back?

**Comment by levicki** [comment]
@hackel

I think that if your change is only two lines of actual code then it should only change the code, not the formatting, regardless of how broken it may be. If this is accepted as it is, someone looking at it later would have to do a full compare to make sure all the rest of the changes are cosmetic in nature.

TL;DR - if you want to fix the formatting as well, submit that as another pull request with appropriate title, but please put those spaces back in for now if that means this change will be merged and released.

**Comment by bdurrer** [comment]
@codata-services that might be, but this PR is laying around for five months now 😢

**Comment by DanCech** [comment]
@hackel You can revert the whitespace-only changes via this process: http://blog.ranjansakalley.com/2015/09/cleaning-up-whitespace-changes-in-a-pull-request.html

**Comment by luckyrat** [comment]
Per the announcement at http://keefox.org/news/detail/2017/03/26/changes-to-keefox-in-2017 I won't be developing KeeFox version 1.x beyond what is essential to ensure a smooth migration to version 2.0.

I'm aware of this PR and related issues and will try to add an equivalent feature to KeeFox 2.0.

**Comment by DanCech** [comment]
@luckyrat so you're saying that this (imho quite important) bugfix won't make it into 1.7?

**Comment by levicki** [comment]
@luckyrat

I am really disappointed by your decision to close this.

You have no obligation to fix any reported issues. However, if someone does it for you after months of people complaining about it left and right, and you outright reject it, then you don't exactly appear friendly towards your user base.

If we have to wait for KeeFox 2.0 for this bug to be fixed we may all as well just use some other extension, thankfully yours is not the only one.

**Comment by luckyrat** [comment]
I'm not outright rejecting this but I simply can't risk a change like this in the critically important 1.7 release. The same argument holds for everything except critical bugs and security problems. I'm sorry that you're disappointed but all I can suggest is to try out the KeeFox 2.0 beta when it's available later in the year.

**Comment by DanCech** [comment]
@luckyrat can you elaborate on "a change like this"?  I would argue that the issue constitutes a "critical bug" since there are many sites (and more every day) where this issue renders KeeFox next to unusable because it currently does not properly simulate filling the form fields.

The fix itself is trivial, dispatching a single browser "change" event to properly simulate the field being filled by a user.  If the white-space changes in this patch are an issue I would be happy to create a PR that does not include them, so the patch would look like https://github.com/kee-org/KeeFox/pull/739/files?w=1

If the 2.0 release was imminent then I wouldn't be arguing for inclusion in 1.7, but asking users to wait until "roughly November" for a one-line patch is a tough pill to swallow.

**Comment by levicki** [comment]
@DanCech Well said.

@luckyrat I am capable of editing that one line myself in downloaded XPI file before installing it, and frankly, that would have been my preferred solution before Firefox 52.0.

However, from 52.0 onwards it is not possible to install unsigned XPIs and there is no user override. I could only do one of the following:

1. Fork KeeFox, change one line, change all metadata so it doesn't clash with your extension, register AMO account, submit XPI for signing, wait, and then apply the same patch later for 1.7 and 1.8 releases.

2. Run ESR or unbranded Firefox release which allows disabling the signing requirement and then be exposed to malware risk all the time.

3. Switch to another extension which does what is necessary.

Option 1 is too much effort for those who are not already registered extension developers, option 2 is obviously out of the question, so it seems that only option 3 remains unless you reconsider.

**Comment by hackel** [comment]
I can understand the logic behind wanting 1.7 to be stable, and am glad something like this will make it into 2.0.  Just in case you change your mind, though, I created #775 with the whitespace un-fixed.  Did you have any specific stability concerns?

In the meantime, it would be interesting to see if we can get chromeIPass to work under Firefox as a WebExtension.
